CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.6)
+cmake_minimum_required(VERSION 2.8.12)
```

Comment:
After I get the rest of this WIP branch stabilized, I was planning on changing [Travis to use](https://github.com/girder/girder/blob/master/.travis.yml#L26) the same version as whatever we decide the minimum should be (it's using 3.1 now). That way, if a commit relies upon a newer CMake feature, the test will fail.

Note that [Ubuntu 12.04 uses CMake 2.8.7](http://packages.ubuntu.com/precise/cmake). I doubt we'll ever see anything older than that in the wild, and it's not unreasonable to ask for a newer just to run tests (it's not needed for deployment). The current requirement of CMake 2.8.6 seems kind of arbitrary (unless there's a good, undocumented reason).